src/Symfony/Component/DependencyInjection/Dumper/YamlDumper.php

```diff
@@ -247,10 +249,11 @@ private function dumpCallable($callable)
      */
     private function dumpValue($value)
     {
-        if ($value instanceof IteratorArgument) {
-            $value = array('=iterator' => $value->getValues());
-        } elseif ($value instanceof ClosureProxyArgument) {
-            $value = array('=closure_proxy' => $value->getValues());
+        if ($value instanceof IteratorArgument || $value instanceof ClosureProxyArgument) {
+            $argument = clone $value;
+            $argument->setValues($this->dumpValue($argument->getValues()));
```

Comment:
this looks like a code smell to me:
this creates an instance of `IteratorArgument` that is filled with a non usable state. Let's imagine that "setValues" does some validation, which it could very well do - that would make the current design fail.
I think we need a more robust design here, that preserves the semantic of objects.